crates/noirc_frontend/src/ast/expression.rs

```diff
@@ -364,11 +364,19 @@ pub struct FunctionDefinition {
     pub body: BlockExpression,
     pub span: Span,
     pub where_clause: Vec<TraitConstraint>,
-    pub return_type: UnresolvedType,
+    pub return_type: FunctionReturnType,
     pub return_visibility: noirc_abi::AbiVisibility,
     pub return_distinctness: noirc_abi::AbiDistinctness,
 }
 
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum FunctionReturnType {
+    /// Returns type is not specified.
+    Default(Span),
+    /// Everything else.
+    Ty(UnresolvedType, Span),
```

Comment:
Can we break `return_type` into two separate fields for the old `return_type: Option<UnresolvedType>` and a new `return_type_span: Span`?

Reply:
what about FuncMeta?

Reply:
Hmm, I was thinking`FuncMeta` should have access to the return type already through its `typ` field and we could add a return type span field, but we'd need something to indicate whether a return type was specified or not. I'll just resolve this since it was a minor point anyway.